src/excmds.ts

```diff
@@ -6208,3 +6209,27 @@ export async function elementunhide() {
     elem.className = elem.className.replace("TridactylKilledElem", "")
 }
 // vim: tabstop=4 shiftwidth=4 expandtab
+
+/**
+ * Move the current TST tree to be just in front of the tab specified.
+ */
+//#background
+export async function tstmove(tabId: string) {
+    treestyletab.moveTreeBefore(Number(tabId))
```

Comment:
if we made `tstmove` operate on Tridactyl tab indices we wouldn't need such extensive changes to the completions - is there a downside to that?

```suggestion
export async function tstmove(tabindex: string) {
    const id = await idFromIndex(tabindex)
    treestyletab.moveTreeBefore(id)
```

Reply:
The TST itself works with IDs: https://github.com/piroor/treestyletab/wiki/API-for-other-addons#move-tree-to-different-position

What we can try doing is to pass an index here and calculate tab ID from it to pass to TST. Is this what you're proposing? If so, I can try reworking like this.

Reply:
yes, that's what I'm proposing :)

it should simplify the completion changes enormously - all you would need to do is add tstmove etc to the commands which get tab completions

Reply:
Ok, sounds reasonable to me. I'll try this rework during my spare time this week and ping you in this thread once ready. Thanks!

Reply:
Thanks! it shouldn't be much work - your functions in excmds.ts would gain a single line like my patch above, the completions code would get reverted and you'd add the command names to the array that triggers the Tab completions 

if you get stuck let me know

Reply:
@bovine3dom applied your suggestion. The code looks much cleaner now, thanks!